.codespell/ignore-words.txt

```diff
@@ -22,8 +29,11 @@ nd
 ned
 re-usable
 re-use
+re-used
+re-using
 recuse
 reenable
 referencable
 therefor
+tim
```

Comment:
Do we need both this and the uppercase one above?

Reply:
Let's remove this line, and these ones above, they don't seem to be needed:

Chang
Damon
Forman
Manuel
Tim

Reply:
They are if you run with `--builtin clear,rare,names` - but happy to take them out. I will add the exact codespell command I used to the PR text.

Reply:
By the way, we have a `make spellcheck` command, and it's run (but not enforced) on the CI. Here's `main`:

https://github.com/python/peps/actions/runs/10443074163/job/28916071128#step:5:98

Reply:
> Do we need both this and the uppercase one above?

Yes, if we want to use the `names` dict. Not strictly necessary, but I've excluded all the false positives, so no harm in using it for now?

Reply:
I ran with extra dicts:

```
builtin = clear,rare,names
```

Not sure it's worth it - but the excludes above are for those more exhaustive settings.

Happy to remove if you prefer to keep it minimal - let me know!

Reply:
Yes please, let's not spellcheck names :)

Reply:
@hugovk I'm sorry I should have been clearer - `--builtin names` does _not_ check for typos in _names_, it checks for typos that _could_ be names as well - so typos with a relatively higher false positivity rate. That's why they aren't enabled by default. However, once one has added the false positives to the ignore-list, it means more typos will be caught.

Does that make sense? I'm still happy to remove if you prefer to keep things simple and the ignore lists shorter. Just wanted to clarify the misunderstanding.

```
 --builtin BUILTIN-LIST
                        comma-separated list of builtin dictionaries to include (when "-D -" or no "-D" is passed). Current options are:
                        - 'clear' for unambiguous errors
                        - 'rare' for rare (but valid) words that are likely to be errors
                        - 'informal' for making informal words more formal
                        - 'usage' for replacing phrasing with recommended terms
                        - 'code' for words from code and/or mathematics that are likely to be typos in other contexts (such as uint)
 ---SEE HERE ----->     - 'names' for valid proper names that might be typos
                        - 'en-GB_to_en-US' for corrections from en-GB to en-US
                        The default is 'clear,rare'.
```

Reply:
Is there some documentation for this?

Do I understand right: with `names` enabled, by default  it will suggest Chang -> change, Damon -> daemon, Manuel -> manual, Tim -> time, and so on? (And also for chang -> change etc.) 

So we need to add those names to an exclude list to prevent this? 

Can codespell instead ignore those uppercase names ("Chang") and only suggest for lowercase ("chang")? 

And the default codespell behaviour is not to suggest changing potential names like C/chang -> change? 

In any case, I think it would be better to avoid "correcting" people's names, especially as we have a large, international group of PEP authors.

Reply:
>Is there some documentation for this?

It's documented at least in the help output:

```
   -L WORDS, --ignore-words-list WORDS
                         comma-separated list of words to be ignored by
                         codespell. Words are case sensitive based on how they
                         are written in the dictionary file.
```

Maybe @DimitriPapadopoulos knows whether something else exists?

>Do I understand right: with names enabled, by default it will suggest Chang -> change, Damon -> daemon, Manuel -> manual, Tim -> time, and so on? (And also for chang -> change etc.)
That's correct, with names enabled, any name that appears in this list will have to be separately excluded for both upper and lower case. I also tested this again just now:
```
defaults/constraint.nwk:832: Chang ==> Change
defaults/constraint.nwk:833: chang ==> change
defaults/constraint.nwk:834: Manuel ==> Manual
defaults/constraint.nwk:835: manuel ==> manual
```

These are the extra typos fixed by names - might not be worth the hassle:
https://github.com/codespell-project/codespell/blob/master/codespell_lib/data/dictionary_names.txt

>So we need to add those names to an exclude list to prevent this?
Correct, I have added all the names that appear in this repo to the ignore list.

>Can codespell instead ignore those uppercase names ("Chang") and only suggest for lowercase ("chang")?
I don't think codespell supports this right now, but I can make a feature request and have a look at it. Sounds like a very reasonable request to ignore upper-case names and only suggest for lower-case ones.

>And the default codespell behaviour is not to suggest changing potential names like C/chang -> change?

That's correct. The default is `clear,rare`. I now added `code` and `names` and excluded the false positives triggered by it. We don't need to do this. We can use defaults if you prefer.

>In any case, I think it would be better to avoid "correcting" people's names, especially as we have a large, international group of PEP authors.

Given that codespell was entirely ignored for a few years and any PR will be reviewed by people manually, it's unlikely we would be changing anyone's name - unless you suggest to start requiring clean codespell in CI - in which case I agree, we should reduce false positive risk as much as possible so as not to bother contributors. If that's the path you're planning to go down (probably not a bad idea given how many typos accrue over time) it would make sense to run at most the defaults, maybe also exclude `rare`.

Reply:
To not waste your time, here's a proposal for going back to just the plain defaults: https://github.com/corneliusroemer/peps/pull/1

Happy to merge this into this branch

Reply:
I'd second just using the defaults.

Reply:
Done, I've merged - I think that should make everyone happy, let me know if not